Refactor `chip::Test` namespace to `chip::Testing`

LyudmilaKostanyan opened the pull request:
#### Summary

Refactored the `chip::Test` namespace to `chip::Testing` across the codebase to consolidate testing-related utilities and resolve namespace inconsistencies.

**Note:** These are minimal changes required to make the code compile successfully.

The refactoring involved:
- Updating namespace declarations from `chip::Test` to `chip::Testing`
- Fixing all references to classes and functions in test files
- Correcting namespace structure to avoid nested namespace issues

#### Related issues

#42186

#### Testing

- Verified compilation succeeds
- No functional changes, only namespace refactoring

LyudmilaKostanyan committed:
Initial commit

LyudmilaKostanyan committed:
Refactor test context usage in various cluster tests to use `chip::Testing` namespace instead of `chip::Test`.

LyudmilaKostanyan committed:
Refactor test cases to use `SetMockNodeConfig` directly from the `Testing` namespace

LyudmilaKostanyan committed:
Refactor ClusterTester.h

LyudmilaKostanyan committed:
Update test files to consistently use `chip::Testing` namespace for test context and cluster tester instances.

LyudmilaKostanyan committed:
Update references from `chip::Test` to `chip::Testing` across multiple files for consistency in test context usage.

LyudmilaKostanyan committed:
Refactor test files to replace all instances of `chip::Test` with `chip::Testing` for improved consistency and clarity in test context usage.

LyudmilaKostanyan committed:
Refactor AppTestContext.h

LyudmilaKostanyan committed:
Refactor test-interaction-model-api.h for improved readability and consistency in formatting.

LyudmilaKostanyan committed:
Refactor MessagingContext.h to improve code readability

LyudmilaKostanyan committed:
Refactor test files to replace all instances of `chip::Test` with `chip::Testing` for improved consistency and clarity in test context usage.

LyudmilaKostanyan committed:
Refactor LoopbackTransport and IOContext classes for improved readability

LyudmilaKostanyan committed:
Refactor test files to consistently use `chip::Testing` namespace, enhancing clarity and uniformity in test context usage across multiple test files.

LyudmilaKostanyan committed:
Refactor ICDConfigurationData and related test access class to consistently use `chip::Testing` namespace, enhancing clarity and uniformity in test context usage.

LyudmilaKostanyan committed:
Update test files to consistently use `chip::Testing` namespace, enhancing clarity and uniformity in test context usage across various components.

LyudmilaKostanyan committed:
Refactor test files to consistently use `chip::Testing` namespace, improving clarity and uniformity in test context usage across DataModelFixtures, TestRead, and TestCodeDrivenDataModelProvider components.

LyudmilaKostanyan committed:
Refactor test files to replace all instances of `chip::Test` with `chip::Testing`, ensuring consistency and clarity in test context usage across various components.

LyudmilaKostanyan committed:
Refactor attribute-storage.cpp to replace `namespace Test` with `namespace chip::Testing`, ensuring consistency in test context usage across the file.

restyled-commits committed:
Restyled by whitespace

restyled-commits committed:
Restyled by clang-format

LyudmilaKostanyan committed:
Remove unnecessary `using namespace chip::Testing;` from TestRead.cpp, improving clarity in the test context.

LyudmilaKostanyan committed:
Update TestMessagingLayer to use `chip::Testing` namespace for consistency in test context usage.

restyled-commits committed:
Restyled by clang-format

Elen777300 commented on src/app/tests/TestAclEvent.cpp:
```diff
@@ -62,17 +62,17 @@ chip::EventId kTestEventIdCritical = 2;
         MockEndpointConfig(kTestEndpointId, {
 
-            MockClusterConfig(chip::Test::kTestDeniedClusterId2,
+            MockClusterConfig(chip::Testing::kTestDeniedClusterId2,
```
This is for the whole file. Please, revisit.

LyudmilaKostanyan replied:
There is no `using namespace chip::Testing`

Elen777300 commented on src/protocols/secure_channel/tests/TestPASESession.cpp:
```diff
@@ -101,7 +101,7 @@ class TestPASESession : public chip::Test::LoopbackMessagingContext
 };
 
-class PASETestLoopbackTransportDelegate : public Test::LoopbackTransportDelegate
+class PASETestLoopbackTransportDelegate : public chip::Testing::LoopbackTransportDelegate
```
```suggestion
class PASETestLoopbackTransportDelegate : public Testing::LoopbackTransportDelegate
```

LyudmilaKostanyan committed:
Refactor test files to consistently use `DispatchCount` and `GetLastDispatchPath` functions, enhancing clarity and uniformity in test context usage across TestServerClusterShim and TestICDManager components.

LyudmilaKostanyan commented on examples/common/server-cluster-shim/tests/TestServerClusterShim.cpp:
```diff
@@ -506,39 +506,39 @@ TEST_F(TestServerClusterShim, EmberInvokeTest)
         ASSERT_EQ(cluster_with_valid_path.InvokeCommand(kInvokeRequest, tlvReader, /* handler = */ nullptr), std::nullopt);
-        EXPECT_EQ(chip::Test::DispatchCount(), kDispatchCountPre + 1); // one dispatch expected
-        EXPECT_EQ(chip::Test::GetLastDispatchPath(), kCommandPath);    // for the right path
+        EXPECT_EQ(chip::Testing::DispatchCount(), kDispatchCountPre + 1); // one dispatch expected
```
Done

LyudmilaKostanyan commented on src/app/icd/server/tests/TestICDManager.cpp:
```diff
@@ -1051,7 +1051,7 @@ TEST_F(TestICDManager, TestICDStateObserverOnICDModeChange)
 TEST_F(TestICDManager, TestICDStateObserverOnICDModeChangeOnInit)
 {
-    ICDConfigurationDataTestAccess privateIcdConfigData(&ICDConfigurationData::GetInstance());
+    Testing::ICDConfigurationDataTestAccess privateIcdConfigData(&ICDConfigurationData::GetInstance());
```
Done

LyudmilaKostanyan commented on src/app/icd/server/tests/TestICDManager.cpp:
```diff
@@ -1149,7 +1149,7 @@ TEST_F(TestICDManager, TestICDStateObserverOnTransitionToIdleModeGreaterActiveMo
 TEST_F(TestICDManager, TestICDStateObserverOnTransitionToIdleModeEqualActiveModeDuration)
 {
-    ICDConfigurationDataTestAccess privateIcdConfigData(&ICDConfigurationData::GetInstance());
+    Testing::ICDConfigurationDataTestAccess privateIcdConfigData(&ICDConfigurationData::GetInstance());
```
Done

LyudmilaKostanyan commented on src/app/icd/server/tests/TestICDManager.cpp:
```diff
@@ -1002,7 +1002,7 @@ TEST_F(TestICDManager, TestICDStateObserverOnEnterActiveMode)
 {
     typedef ICDListener::ICDManagementEvents ICDMEvent;
-    ICDConfigurationDataTestAccess privateIcdConfigData(&ICDConfigurationData::GetInstance());
+    Testing::ICDConfigurationDataTestAccess privateIcdConfigData(&ICDConfigurationData::GetInstance());
```
Done

LyudmilaKostanyan committed:
Refactor test-ember-api.h to improve namespace formatting and enhance clarity in the test context usage.

LyudmilaKostanyan committed:
Refactor TestExchange to use `Testing::LoopbackMessagingContext` directly, enhancing consistency in test context usage.

LyudmilaKostanyan commented on src/messaging/tests/TestExchange.cpp:
```diff
@@ -45,14 +45,14 @@ using namespace chip::Messaging;
         ASSERT_EQ(psa_crypto_init(), PSA_SUCCESS);
 #endif
-        chip::Test::LoopbackMessagingContext::SetUp();
+        chip::Testing::LoopbackMessagingContext::SetUp();
```
Done

LyudmilaKostanyan committed:
Update src/app/util/mock/attribute-storage.cpp

Co-authored-by: Elen777300 <43118262+Elen777300@users.noreply.github.com>

restyled-commits committed:
Restyled by clang-format

LyudmilaKostanyan committed:
Refactor TestPASESession to directly use `Testing::LoopbackMessagingContext`, improving consistency in test context usage.

LyudmilaKostanyan committed:
Refactor TestCodegenModelViaMocks to use `Protocols::InteractionModel::Status` directly, improving consistency in status handling within tests.

LyudmilaKostanyan committed:
Refactor MockNodeConfig to improve code consistency by replacing `internal::kDefaultStringSize` with `kDefaultStringSize` and streamline constructor initializations in MockClusterConfig and MockEndpointConfig.

LyudmilaKostanyan committed:
Refactor DataModelFixtures to use `Testing::MockClusterId` and `Testing::MockAttributeId` directly, enhancing consistency in test fixture definitions.

LyudmilaKostanyan committed:
Refactor TestAclEvent to improve namespace usage and enhance consistency by removing redundant `chip::` prefixes, streamlining access to types and functions within the app namespace.

restyled-commits committed:
Restyled by clang-format

LyudmilaKostanyan committed:
Remove redundant namespace declaration in attribute-storage.cpp to improve code clarity and maintain consistency in namespace usage.

LyudmilaKostanyan committed:
Merge branch 'master' into RefactoringNamespaces

LyudmilaKostanyan committed:
Merge branch 'master' into RefactoringNamespaces

LyudmilaKostanyan committed:
Refactor test namespaces for consistency across network commissioning and occupancy sensing clusters.

Update usage of TestServerClusterContext and ClusterTester to the Testing namespace in relevant test files.

LyudmilaKostanyan committed:
Refactor `DataModelFixtures.h `

restyled-commits committed:
Restyled by clang-format

LyudmilaKostanyan committed:
Remove unused includes in `TestResourceMonitoringCluster.cpp`

LyudmilaKostanyan committed:
Merge branch 'master' into RefactoringNamespaces

LyudmilaKostanyan committed:
Merge branch 'master' into RefactoringNamespaces

LyudmilaKostanyan committed:
Merge branch 'master' into RefactoringNamespaces

LyudmilaKostanyan committed:
Refactor namespace usage in `Group Key Management Cluster` tests
